pkg/worker/k8srunner.go

```diff
@@ -252,17 +252,27 @@ func (r k8sStepRunner) waitForPodModifiedFunc(ctx context.Context, podMeta metav
 	}
 	defer w.Stop()
 	for ev := range w.ResultChan() {
-		pod := ev.Object.(*v1.Pod)
-		switch ev.Type {
-		case watch.Modified:
-			ok, err := f(pod)
-			if err != nil {
-				return err
-			} else if ok {
-				return nil
+		switch ev.Object.(type) {
+		case *v1.Pod:
+			pod := ev.Object.(*v1.Pod)
+			switch ev.Type {
+			case watch.Modified:
+				ok, err := f(pod)
+				if err != nil {
+					return err
+				} else if ok {
+					return nil
+				}
+			case watch.Deleted:
+				return fmt.Errorf("pod was removed: %v", pod.Name)
+			}
+		case *metav1.Status:
+			if errors.Is(ctx.Err(), context.Canceled) {
+				return fmt.Errorf("context was cancelled for pod: %v", podMeta.Name)
```

Comment:
nit: Returned error should contain the context error IMO.

And as the context error already contains the text `"context canceled"`, then you could leave that out from the new error

```suggestion
				return fmt.Errorf("watching pod: %s: %w", podMeta.Name, ctx.Err())
```

Reply:
Nice 👍🏻